shared/actions/favorite.js

```diff
@@ -24,71 +24,71 @@ function toggleShowIgnored (isPrivate: boolean): FavoriteToggleIgnored {
   return {type: Constants.favoriteToggleIgnored, payload: {isPrivate}, error: false}
 }
 
+function favoriteList (): FavoriteList {
+  return {type: Constants.favoriteList, payload: undefined}
+}
+
+function favoriteFolder (path: string): FavoriteAdd {
+  return {type: Constants.favoriteAdd, payload: {path}}
+}
+
+function ignoreFolder (path: string): FavoriteIgnore {
+  return {type: Constants.favoriteIgnore, payload: {path}}
+}
+
 const injectMeta = type => f => { f.meta = type }
 
 const _jsonToFolders = (json: Object, myKID: any) => {
   const folderSets = [json.favorites, json.ignored, json.new]
+  const fillFolder = folder => {
+    folder.waitingForParticipantUnlock = []
+    folder.youCanUnlock = []
 
-  folderSets.forEach(folders => {
```

Comment:
this part of the diff looks kinda crappy in github but its just hoisting the closure into a helper